src/main/java/mate/jdbc/dao/DriverDaoImpl.java

```diff
@@ -0,0 +1,118 @@
+                PreparedStatement getStatement =
+                        connection.prepareStatement(getRequest)) {
+            getStatement.setLong(1, id);
+            ResultSet resultSet = getStatement.executeQuery();
+            Driver driver = null;
+            if (resultSet.next()) {
+                driver = setValuesFromDB(resultSet);
+            }
+            return Optional.of(driver);
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get driver for id: "
+                    + id + " from DB", e);
+        }
+    }
+
+    @Override
+    public List<Driver> getAll() {
+        String getAllRequest = "SELECT * FROM drivers WHERE is_deleted = false";
+        List<Driver> allDrivers = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                Statement getAllStatement = connection.createStatement()) {
+            ResultSet resultSet =
+                    getAllStatement.executeQuery(getAllRequest);
+            while (resultSet.next()) {
+                allDrivers.add(setValuesFromDB(resultSet));
+            }
+            return allDrivers;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t get all driver from DB", e);
+        }
+    }
+
+    @Override
+    public Driver update(Driver driver) {
+        String updateRequest = "UPDATE drivers SET name = ?, "
+                + "licenseNumber = ? WHERE id = ? and is_deleted = false;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createStatement =
+                        connection.prepareStatement(updateRequest)) {
+            createStatement.setString(1, driver.getName());
+            createStatement.setString(2, driver.getLicenseNumber());
+            createStatement.setLong(3, driver.getId());
+            if (createStatement.executeUpdate() > 0) {
+                return driver;
+            }
+            throw new RuntimeException("No such driver " + driver.getId());
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t update driver for name: "
+                    + driver.getName() + " from DB", e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteRequest = "UPDATE drivers SET is_deleted = true where id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement createStatement =
+                        connection.prepareStatement(deleteRequest)) {
+            createStatement.setLong(1, id);
+            return createStatement.executeUpdate() > 0;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t delete driver for id: "
+                    + id + " from DB", e);
+        }
+    }
+
+    public Driver setValuesFromDB(ResultSet resultSet) throws SQLException {
```

Comment:
should this method be public? also, let's make the name more informative

Reply:
changed